### PR TITLE
L2-958: Manage error project not found

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

User should be redirected to launchpad when trying to access details of a non-existing project.

# Changes

* Changes in the ProjectDetail page to manage edge case. Specifically in mapProjectDetail.

# Tests

* Provide test for not found or wrong canister id in the url of Project Detail.

### CODE REVIEW

**Comment by lmuntaner** [comment]
New UX:

Both for a valid canister id that is not present, and an invalid canister id. Both are the same for the user.

https://user-images.githubusercontent.com/4550653/186099861-c84f7f7e-5fb2-495f-a25a-49d1cdc7c318.mov

**Comment by lmuntaner** [comment]
@peterpeterparker @mstrasinskis please review

File: frontend/src/routes/ProjectDetail.svelte [hunk 1]
```diff
@@ -120,6 +135,9 @@
       }
 
       const rootCanisterId = routePathRootCanisterId(path);
+      if (rootCanisterId === undefined) {
```
Comment (peterpeterparker) [inline frontend/src/routes/ProjectDetail.svelte:138]: we need that checks?

File: frontend/src/routes/ProjectDetail.svelte [hunk 2]
```diff
@@ -146,17 +164,25 @@
...
   $: layoutTitleStore.set($projectDetailStore?.summary?.metadata.name ?? "");
 
-  let loadingSummary: boolean;
-  $: loadingSummary = isNullish($projectDetailStore.summary);
-  let loadingSwapState: boolean;
-  $: loadingSwapState = isNullish($projectDetailStore.swapCommitment);
-
-  // TODO(L2-838): if error redirect to launchpad and display error there
+  let loading: boolean;
+  $: loading = $snsSummariesStore.length === 0;
```
Comment (peterpeterparker) [inline frontend/src/routes/ProjectDetail.svelte:168]: previously the swap commitment was also considered for the "loading" check. any particular reason why not considering it anymore?

File: frontend/src/routes/ProjectDetail.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/routes/ProjectDetail.svelte:138]: I avoided the check inside `mapProjectDetail` this way. But this is an edge case, the check `isRoutePath({ path: AppPath.ProjectDetail, routePath: path }` already makes sure that there will be such a `rootCanisterId`. Or that at least it will be defined.

File: frontend/src/routes/ProjectDetail.svelte (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/routes/ProjectDetail.svelte:168]: I thought one was enough, but I will add it.

Comment (lmuntaner) [reply frontend/src/routes/ProjectDetail.svelte]: done

File: frontend/src/routes/ProjectDetail.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/routes/ProjectDetail.svelte:138]: oki doki, thx for the feedback

**Review by peterpeterparker — APPROVED** [review]
LGTM, Thx
